gitlabform/gitlab/python_gitlab.py

```diff
@@ -0,0 +1,61 @@
+import functools
+
+from gitlab import Gitlab, GitlabGetError
+from gitlab.base import RESTObject, RESTObjectList
+from gitlab.v4.objects import Group
+
+
+# Extends the python-gitlab class to add convenience wrappers for common functionality used within gitlabform
+class PythonGitlab(Gitlab):
+    def get_user_id(self, username) -> int:
+        user = self.get_user_by_username(username)
+        return user.id
+
+    def get_group_id(self, groupname) -> int:
+        groups = self.get_group_by_groupname(groupname)
+        return groups.id
+
+    def get_group_by_name(self, groupname: str) -> Group:
+        group = self.get_group_by_groupname(groupname)
+        return self.groups.get(group.id)
+
+        #  Uses "LIST" to get a group by groupname, to get the full Group object, call get using the group's id
+
+    @functools.lru_cache()
```

Comment:
Thanks for caring about performance and keeping the cache here, @TimKnight-DWP! 😊 👍

Reply:
Thanks 👍